### PR TITLE
Event registration

### PR DESCRIPTION
**Pull request by ZAINABFATIMA0**
Added registration in events through email verification,

### CODE REVIEW

**Review by Hamza-Chaudhary65 — APPROVED** [review]

File: eventify/settings.py [hunk 1]
```diff
@@ -160,3 +160,12 @@
 }
 
 AUTH_USER_MODEL = 'users.User'
+
+CELERY_BROKER_URL = 'redis://127.0.0.1:6379'
```
Comment (sheralim012) [inline eventify/settings.py]: Value for this should be coming from env variable

File: Readme.md [hunk 2]
```diff
@@ -41,3 +41,61 @@ REDIS_HOST=localhost
...
+Ensure you have Celery installed on your system. If Celery is not installed, follow the below instructions.
+
+### Insallation
+
+Configure Celery in settings.py:
+**Run the following commands in command line:**
+
+```bash
+pip install celery
+```
```
Comment (sheralim012) [inline Readme.md]: is this instruction necessary?, as the codebase has already this info

File: Readme.md [hunk 3]
```diff
@@ -41,3 +41,61 @@ REDIS_HOST=localhost
...
+### Configuration
+We will be using redis as a broker for cwlwery which we have already installed and configured above.
+Add the following Celery settings in your settings.py:
```
Comment (sheralim012) [inline Readme.md]: the value for celery broker url should come from env variable

File: users/serializer.py [hunk 4]
```diff
@@ -29,3 +34,21 @@ def create(self, validated_data):
...
         return user
+
+
+class RegistrationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Registration
+        fields = '__all__'
+
+    def create(self, validated_data):
+        registration, created = Registration.objects.update_or_create(
```
Comment (sheralim012) [inline users/serializer.py:54]: add validation for if registration form is open

File: users/serializer.py [hunk 5]
```diff
@@ -29,3 +34,21 @@ def create(self, validated_data):
...
+            email=validated_data.get('email'),
+            event=validated_data.get('event'),
+            defaults={
+                'otp': random.randint(100000, 999999),
+                'otp_expiry': timezone.now() + timedelta(minutes=5)
```
Comment (sheralim012) [inline users/serializer.py]: timedelta value should be configurable

File: events/views.py [hunk 6]
```diff
@@ -38,3 +41,49 @@ def list_category(request):
...
         serializer = CategorySerializer(categories, many=True)
         return Response(serializer.data)
+
+@api_view(['GET'])
+@permission_classes([AllowAny])
+def get_event(request, pk):
+    event = get_object_or_404(Event, pk=pk)
+    serializer = EventSerializer(event)
+    return Response(serializer.data)
+
```
Comment (sheralim012) [inline events/views.py:52]:
```suggestion
 
 
```

File: events/views.py [hunk 7]
```diff
@@ -38,3 +41,49 @@ def list_category(request):
...
+@api_view(['POST'])
+@permission_classes([AllowAny])
+def register_for_event(request, pk):
+  
+    serializer = RegistrationSerializer(
+        data={
+            'email': request.data['email'],
+            'event': pk
```
Comment (sheralim012) [inline events/views.py]: why not pass whole data object from request?

File: events/views.py [hunk 8]
```diff
@@ -38,3 +41,49 @@ def list_category(request):
...
+        }
+    )
+    serializer.is_valid(raise_exception=True)
+    serializer.save()
+
+    return Response({"message": "Registration created successfully. Email is not verified"})
```
Comment (sheralim012) [inline events/views.py]: update msg to indicate that check your email for OTP

File: events/views.py [hunk 9]
```diff
@@ -38,3 +41,49 @@ def list_category(request):
...
+    )
+    serializer.is_valid(raise_exception=True)
+
+    registration = get_object_or_404(
+        Registration, 
+        email=serializer.validated_data['email'], 
+        event_id=pk
+    )
+    registration.is_verified = True
+    registration.otp = None
```
Comment (sheralim012) [inline events/views.py:81]: reset opt expiry as well

File: events/views.py [hunk 10]
```diff
@@ -38,3 +41,49 @@ def list_category(request):
...
+    registration.save()
```
Comment (sheralim012) [inline events/views.py]: if small set of fields are updated then it's recommended to use update_fields in save as it's more performant

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**